app/src/main/java/com/github/orkest/Model/AudioRecognition.kt

```diff
@@ -0,0 +1,174 @@
+package com.github.orkest.Model
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import androidx.annotation.RequiresPermission
+import com.github.orkest.Constants
+import com.shazam.shazamkit.AudioSampleRateInHz
+import com.shazam.shazamkit.MatchResult
+import com.shazam.shazamkit.ShazamKit
+import com.shazam.shazamkit.ShazamKitResult
+import com.shazam.shazamkit.SignatureGenerator
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import java.nio.ByteBuffer
+import java.util.concurrent.CompletableFuture
+
+class AudioRecognition {
+
+    companion object {
+
+         val catalog =
+                ShazamKit.createCustomCatalog().apply {
+
+                }
+
+        private fun Int.toByteAllocation(): Int {
+            return when (this) {
+                AudioFormat.ENCODING_PCM_16BIT -> 2
+                else -> throw IllegalArgumentException("Unsupported encoding")
+            }
+        }
+
+        private fun ByteBuffer.putTrimming(byteArray: ByteArray) {
+            if (byteArray.size <= this.capacity() - this.position()) {
+                this.put(byteArray)
+            } else {
+                this.put(byteArray, 0, this.capacity() - this.position())
+            }
```

Comment:
Looks good to me but a little bit of comment could be appreciated :)

Reply:
Yes sorry, I added this sample recording just to implement a dummy communication with Shazam so I didn't really spend a lot of time on this part of the code, it will be enhanced in next sprint